src/cli/cli.cc

```diff
@@ -77,8 +77,9 @@ StatusCode CheckIn(AkliteClient &client, const LocalUpdateSource *local_update_s
   return res2StatusCode<CheckInResult::Status>(c2s, cr.status);
 }
 
-StatusCode Install(AkliteClient &client, int version, const std::string &target_name, InstallMode install_mode,
-                   bool force_downgrade, const LocalUpdateSource *local_update_source) {
+static StatusCode pullAndInstall(AkliteClient &client, int version, const std::string &target_name,
```

Comment:
Maybe it makes sense to introduce separate functions `pull` and `install`?

Reply:
IMO, only if e break the current function in some functions that can be reused. Otherwise there is too much duplication between standalone pull and install functions.  I'm taking a look.

Reply:
I spent some time trying to get to a good solution having separate functions, without duplicating much logic. But the resulting code ended up being worse than it is now. So, from my feeling, it is better to keep is as is.

Reply:
That's fine.